orbs/helm/deployable

```diff
@@ -35,13 +35,19 @@ if [ ! -d ".git" ]; then
 fi
 
 last_tag=$(git tag --sort version:refname --list "$pattern" | tail -1)
+# There are no tags at all
 if [ -z "$last_tag" ]; then
     exit 0
 fi
 
 deployed_sha1=$(git rev-parse "$last_tag")
 current_sha1=$(git rev-parse HEAD)
 
-if [ "$(git rev-list $deployed_sha1 | grep $current_sha1)" ]; then
+# List commits behind the latest deployment tag, to see if the current commit is not behind.
+# Exclude the tagged commit itself so we can redeploy an already deployed, and thus tagged,
+# commit.
+# Why isn't it enough to check that "${deployed_sha1}" != "${current_sha1}"?
+if { git rev-list "^${deployed_sha1}" | grep -q "${current_sha1}" ; }
+then
```

Comment:
Maybe the logic is a bit backwards here: what we want is to know if the latest commit taggged as a deployment is reachable from HEAD. In which case we are either ahead of the latest deployment or the latest deployment.

```
 -----o----o----o----o HEAD
       \-<= label
```
But what we check is whether the HEAD is behind the latest tag, which seem much more complicated to check.

The whole code from line 37 onwards, can be replaced with a single check:
```bash
if ! { git rev-list HEAD | grep -q "$(git tag --sort version:refname --format "%(objectname)" --list "deploiment-*" | tail -1)" }
then
    fatal "the current build is older than the deployed one"
fi
```